docs/_developer_onboarding.md

```diff
@@ -0,0 +1,259 @@
+# Developer on boarding
+
+<!--- TOC -->
+
+* [Introduction](#introduction)
+  * [Quick introduction to Matrix](#quick-introduction-to-matrix)
+    * [Matrix data](#matrix-data)
+      * [Room](#room)
+      * [Event](#event)
+    * [Sync](#sync)
+      * [Glossary about syncs](#glossary-about-syncs)
+  * [The Android project](#the-android-project)
+  * [Matrix SDK](#matrix-sdk)
+  * [Application](#application)
+    * [MvRx](#mvrx)
+      * [Behavior](#behavior)
+    * [Epoxy](#epoxy)
+    * [Other frameworks](#other-frameworks)
+  * [Push](#push)
+  * [Dependencies management](#dependencies-management)
+  * [Test](#test)
+  * [Other points](#other-points)
+    * [Logging](#logging)
+    * [Rageshake](#rageshake)
+  * [Tips](#tips)
+* [Happy coding!](#happy-coding)
+
+<!--- END -->
+
+## Introduction
+
+This doc is a quick introduction about the project and its architecture.
+
+It's aim is to help new developers to understand the overall project and where to start developing.
+
+Other useful documentation:
+- all the docs in this folder!
+- the [contributing doc](../CONTRIBUTING.md), that you should also read carefully.
+
+### Quick introduction to Matrix
+
+Matrix website: [matrix.org](https://matrix.org), [discover page](https://matrix.org/discover).
+*Note*: Matrix.org is also hosting a homeserver ([.well-known file](https://matrix.org/.well-known/matrix/client)).
+The reference homeserver (this is how Matrix servers are called) implementation is [Synapse](https://github.com/matrix-org/synapse/). But other implementations exist. The Matrix specification is here to ensure that any Matrix client, such as Element Android and its SDK can talk to any Matrix server.
+
+Have a quick look to the client-server API documentation: [Client-server documentation](https://spec.matrix.org/v1.3/client-server-api/). Other network API exist, the list is here: (https://spec.matrix.org/latest/)
+
+Matrix is an open source protocol. Change are possible and are tracked using [this GitHub repository](https://github.com/matrix-org/matrix-doc/). Changes to the protocol are called MSC: Matrix Spec Change. These are PullRequest to this project.
+
+Matrix object are Json data. Unstable prefixes must be used for Json keys when the MSC is not merged (i.e. accepted). 
+
+#### Matrix data
+
+There are many object and data in the Matrix worlds. Let's focus on the most important and used, `Room` and `Event`
+
+##### Room
+
+`Room` is a place which contains ordered `Event`s. They are identified with their `room_id`. Nearly all the data are stored in rooms, and shared using homeserver to all the Room Member.
+
+*Note*: Spaces are also Rooms with a different `type`.
+
+##### Event
+
+`Events` are item of a Room, where data is embedded.
+
+There are 2 types of Event:
```

Comment:
I'd more write: 
There are 2 types of room Event:
State Events:....
Regular Events:...

You can also have event sent outside of a room, like AccountData

Reply:
OK, updated.